gapic/src/main/com/google/gapid/views/TextureList.java

```diff
@@ -0,0 +1,551 @@
+        createTableColumn(viewer, "Height", Data::getHeight,
+            Comparator.comparingInt(Data::getSortHeight)),
+        createTableColumn(viewer, "Depth", Data::getDepth,
+            Comparator.comparingInt(Data::getSortDepth)),
+        createTableColumn(viewer, "Layers", Data::getLayers,
+            Comparator.comparingInt(Data::getSortLayers)),
+        createTableColumn(viewer, "Levels", Data::getLevels,
+            Comparator.comparingInt(Data::getSortLevels)),
+        createTableColumn(viewer, "Format", Data::getFormat,
+            Comparator.comparing(Data::getFormat)));
+  }
+
+  @Override
+  public Control getControl() {
+    return this;
+  }
+
+  @Override
+  public void reinitialize() {
+    onCaptureLoadingStart(false);
+    updateTextures(true);
+  }
+
+  @Override
+  public void onCaptureLoadingStart(boolean maintainState) {
+    clear();
+  }
+
+  @Override
+  public void onCaptureLoaded(Loadable.Message error) {
+    if (error != null) {
+      clear();
+    }
+  }
+
+  @Override
+  public void onResourcesLoaded() {
+    updateTextures(true);
+  }
+
+  @Override
+  public void onCommandsSelected(CommandIndex path) {
+    updateTextures(false);
+  }
+
+  @Override
+  public void onTextureFollowed(Service.Resource resource) {
+    TableItem[] items = textureTable.getTable().getItems();
+    for (int i = 0; i < items.length; i++) {
+      Data d = (Data)(items[i].getData());
+      if (d.info.getID().equals(resource.getID())) {
+        textureTable.getTable().setSelection(items[i]);
+        break;
+      }
+    }
+  }
+
+  private void updateTextures(boolean resourcesChanged) {
+    if (!models.resources.isLoaded()) {
+      // TODO: show capture loading message
+    } else if (models.commands.getSelectedCommands() == null) {
+      // TODO: show Messages.SELECT_COMMAND
+      clear();
+    } else {
+      // Memorize selection index before disposing image resource.
+      // When comparator is reset, the table is refreshed, and early image disposal will introduce null error.
+      ViewerComparator comparator = textureTable.getComparator();
+      textureTable.setComparator(null);
+      int selection = textureTable.getTable().getSelectionIndex();
+
+      imageProvider.reset();
+
+      Widgets.Refresher refresher = withAsyncRefresh(textureTable);
+      List<Data> textures = Lists.newArrayList();
+      models.resources.getResources(TextureResource).stream()
+          .map(r -> new Data(r.resource, r.deleted))
+          .forEach(data -> {
+            textures.add(data);
+            data.load(models.resources, textureTable.getTable(), refresher);
+          });
+
+      textureTable.setInput(textures);
+      packColumns(textureTable.getTable());
+
+      if (!resourcesChanged && selection >= 0 && selection < textures.size()) {
+        textureTable.getTable().select(selection);
+      }
+      textureTable.setComparator(comparator);
+      updateSelection();
```

Comment:
Also, not a TODO, it should be part of this PR.

Reply:
I added LoadablePanel to show NO_TEXTURES, but we can't show SELECT_TEXTURE here as it would hide the table and it wouldn't be possible to select a texture.